File: src/test/java/com/running/dto/buddy/test/java/cases/contract/BaseClasses.java
package com.running.dto.buddy.test.java.cases.contract;

public interface BaseClasses {

    interface InterfaceWithNoMember {
    }

    interface InterfaceWithAbstractGetter {
        String getName();
    }

    interface InterfaceWithAbstractGetterAndAbstractSetter {
        String getName();

        void setName(String name);
    }

    interface InterfaceWithAbstractProperty {
        String getName();
    }

    abstract static class AbstractClassWithNoMember {
    }

    abstract static class AbstractClassWithAbstractGetter {
        public abstract String getName();
    }

    abstract static class AbstractClassWithAbstractGetterAndAbstractSetter {
        public abstract String getName();

        public abstract void setName(String name);
    }

    abstract static class AbstractClassWithConcreteGetterAndConcreteSetterWithoutField {
        public String getName() {
            return "John Doe";
        }

        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    abstract static class AbstractClassWithConcreteGetterAndConcreteSetterWithField {
        private String _name = "John Doe";

        public String getName() {
            return _name;
        }

        public void setName(String name) {
            this._name = name;
        }
    }

    abstract static class AbstractClassWithAbstractImmutableProperty {
        public abstract String getName();
    }

    abstract static class AbstractClassWithAbstractMutableProperty {
        public abstract String getName();

        public abstract void setName(String name);
    }

    abstract static class AbstractClassWithConcreteMutableProperty {
        private String name = "John Doe";

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    static class ConcreteClassWithNoMember {
    }

    static class ConcreteClassWithGetterAndSetterWithoutField {
        public String getName() {
            return "John Doe";
        }

        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    static class ConcreteClassWithGetterAndSetterWithField {
        private String _name = "John Doe";

        public String getName() {
            return _name;
        }

        public void setName(String name) {
            this._name = name;
        }
    }

    static class ConcreteClassWithConcreteMutableProperty {
        private String name = "John Doe";

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

}

File: src/test/kotlin/com/runninglane/dto/buddy/test/cases/contract/BaseClasses.kt
package com.runninglane.dto.buddy.test.cases.contract

interface InterfaceWithNoMember

interface InterfaceWithAbstractGetter {
    fun getName(): String
}

interface InterfaceWithAbstractGetterAndAbstractSetter {
    fun getName(): String
    fun setName(name: String)
}

interface InterfaceWithDefaultGetter {
    fun getName(): String = "John Doe"
}

interface InterfaceWithDefaultGetterAndAbstractSetter {
    fun getName(): String = "John Doe"
    fun setName(name: String)
}

interface InterfaceWithAbstractProperty {
    val name: String
}

interface InterfaceWithAbstractPropertyAndDefaultGetter {
    val name: String
        get() = "John Doe"
}

abstract class AbstractClassWithNoMember

abstract class AbstractClassWithAbstractGetter {
    abstract fun getName(): String
}

abstract class AbstractClassWithAbstractGetterAndAbstractSetter {
    abstract fun getName(): String
    abstract fun setName(name: String)
}

abstract class AbstractClassWithConcreteGetter {
    fun getName(): String = "John Doe"
}

abstract class AbstractClassWithConcreteSetter {
    fun setName(name: String) { TODO("Not yet implemented") }
}

abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithoutField {
    fun getName(): String = "John Doe"
    fun setName(name: String) { TODO("Not yet implemented") }
}

abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithField {
    private var _name: String = "John Doe"
    fun getName(): String = _name
    fun setName(name: String) { this._name = name }
}

abstract class AbstractClassWithAbstractImmutableProperty {
    abstract val name: String
}

abstract class AbstractClassWithAbstractMutableProperty {
    abstract var name: String
}

abstract class AbstractClassWithConcreteImmutableProperty {
    val name: String = "John Doe"
}

abstract class AbstractClassWithConcreteMutableProperty {
    var name: String = "John Doe"
}

open class ConcreteClassWithNoMember

open class ConcreteClassWithGetter {
    fun getName(): String = "John Doe"
}

open class ConcreteClassWithSetter {
    fun setName(name: String) { TODO("Not yet implemented") }
}

open class ConcreteClassWithGetterAndSetterWithoutField {
    fun getName(): String = "John Doe"
    fun setName(name: String) { TODO("Not yet implemented") }
}

open class ConcreteClassWithGetterAndSetterWithField {
    private var _name: String = "John Doe"
    fun getName(): String = _name
    fun setName(name: String) { this._name = name }
}

open class ConcreteClassWithConcreteImmutableProperty {
    val name: String = "John Doe"
}

open class ConcreteClassWithConcreteMutableProperty {
    var name: String = "John Doe"
}